datafusion/core/src/physical_plan/joins/hash_join.rs

```diff
@@ -852,6 +852,7 @@ fn build_join_indexes(
                             &keys_values,
                             *null_equals_null,
                         )? {
+                            left_indices.append(i);
```

Comment:
Ah good find. I missed the case where this would be used.
Thanks for adding the unit test!